src/main/java/org/sagebionetworks/bridge/exporter/record/BridgeExporterRecordProcessor.java

```diff
@@ -0,0 +1,177 @@
+        this.metricsHelper = metricsHelper;
+    }
+
+    /**
+     * Record filter helper, used to determine which records to filter out, due to request filters or sharing filters.
+     */
+    @Autowired
+    public final void setRecordFilterHelper(RecordFilterHelper recordFilterHelper) {
+        this.recordFilterHelper = recordFilterHelper;
+    }
+
+    /** Record ID source factor. Given the request, this generates the list of record IDs to be exported. */
+    @Autowired
+    public final void setRecordIdSourceFactory(RecordIdSourceFactory recordIdSourceFactory) {
+        this.recordIdSourceFactory = recordIdSourceFactory;
+    }
+
+    /**
+     * Worker manager, which takes export tasks and subtasks generated by this class and manages the workers and
+     * handlers to perform those tasks, and eventually uploads the export results to Synapse.
+     */
+    @Autowired
+    public final void setWorkerManager(ExportWorkerManager workerManager) {
+        this.workerManager = workerManager;
+    }
+
+    /**
+     * Main entry point into Bridge-EX. This process all records for the given request.
+     *
+     * @param request
+     *         request to process records for
+     * @throws IOException
+     *         if we fail to get the record IDs from the record ID factory
+     */
+    public void processRecordsForRequest(BridgeExporterRequest request) throws IOException {
+        LocalDate requestDate = request.getDate();
+        String requestTag = request.getTag();
+        LOG.info("Received request with date=" + requestDate + ", tag=" + requestTag);
+
+        // make task
+        Metrics metrics = new Metrics();
+        File tmpDir = fileHelper.createTempDir();
+        ExportTask task = new ExportTask.Builder().withExporterDate(LocalDate.now(timeZone)).withMetrics(metrics)
+                .withRequest(request).withTmpDir(tmpDir).build();
+
+        Stopwatch stopwatch = Stopwatch.createStarted();
+        try {
+            Iterable<String> recordIdIterable = recordIdSourceFactory.getRecordSourceForRequest(request);
+            for (String oneRecordId : recordIdIterable) {
+                // Count total number of records. Also, log at regular intervals, so people tailing the logs can follow
+                // progress.
+                int numTotal = metrics.incrementCounter("numTotal");
+                if (numTotal % progressReportPeriod == 0) {
+                    LOG.info("Num records so far: " + numTotal + " in " + stopwatch.elapsed(TimeUnit.SECONDS) +
+                            " seconds");
+                }
+
+                // sleep to rate limit our requests to DDB
+                if (delayMillis > 0) {
+                    try {
+                        Thread.sleep(delayMillis);
+                    } catch (InterruptedException ex) {
+                        LOG.error("Record processor interrupted while sleeping: " + ex.getMessage(), ex);
```

Comment:
Do you need to rethrow this, call Thread.currentThread().interrupt(), or anything else since something is trying to kill the thread?